src/Controls/tests/Core.UnitTests/RefreshViewTests.cs

```diff
@@ -167,5 +167,98 @@ public void IsRefreshingEventFires()
 			refreshView.IsRefreshing = true;
 			Assert.True(eventFired);
 		}
+
+		[Fact]
+		public void IsRefreshEnabledDefaultsToTrue()
+		{
+			RefreshView refreshView = new RefreshView();
+			Assert.True(refreshView.IsRefreshEnabled);
+		}
+
+		[Fact]
+		public void IsRefreshEnabledCanBeSetToFalse()
+		{
+			RefreshView refreshView = new RefreshView();
+			refreshView.IsRefreshEnabled = false;
+			Assert.False(refreshView.IsRefreshEnabled);
+		}
+
+		[Fact]
+		public void IsRefreshEnabledPreventsIsRefreshingFromBeingSetToTrue()
+		{
+			RefreshView refreshView = new RefreshView();
+			refreshView.IsRefreshEnabled = false;
+			refreshView.IsRefreshing = true;
+			Assert.False(refreshView.IsRefreshing);
+		}
+
+		[Fact]
+		public void IsRefreshingCanBeSetToFalseWhenIsRefreshEnabledIsFalse()
+		{
+			RefreshView refreshView = new RefreshView();
+			refreshView.IsRefreshing = true;
+			Assert.True(refreshView.IsRefreshing);
+			
+			refreshView.IsRefreshEnabled = false;
+			Assert.False(refreshView.IsRefreshing); // Should be automatically cleared
+		}
+
+		[Fact]
+		public void SettingIsRefreshEnabledToFalseWhileRefreshingStopsRefresh()
+		{
+			RefreshView refreshView = new RefreshView();
+			refreshView.IsRefreshing = true;
+			Assert.True(refreshView.IsRefreshing);
+			
+			refreshView.IsRefreshEnabled = false;
+			Assert.False(refreshView.IsRefreshing);
+		}
+
+		[Fact]
+		public void BothIsEnabledAndIsRefreshEnabledMustBeTrueToAllowRefresh()
+		{
+			RefreshView refreshView = new RefreshView();
+			
+			// Both true - should allow refresh
+			refreshView.IsEnabled = true;
+			refreshView.IsRefreshEnabled = true;
+			refreshView.IsRefreshing = true;
+			Assert.True(refreshView.IsRefreshing);
+			
+			// IsEnabled false - should prevent refresh
+			refreshView.IsRefreshing = false;
+			refreshView.IsEnabled = false;
+			refreshView.IsRefreshing = true;
+			Assert.False(refreshView.IsRefreshing);
+			
+			// IsRefreshEnabled false - should prevent refresh
+			refreshView.IsEnabled = true;
+			refreshView.IsRefreshEnabled = false;
+			refreshView.IsRefreshing = true;
+			Assert.False(refreshView.IsRefreshing);
+		}
```

Comment:
Make this test a Theory and use parameters to control the 3 properties, and 1 to control the expected result. This way we can see if one scenario fails.